pkg/cmd/cli/cmd/set/env.go

```diff
@@ -160,7 +161,11 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 	}
 	// Keep a copy of the original objects prior to updating their environment.
 	// Used in constructing the patch(es) that will be applied in the server.
-	oldObjects, err := resource.AsVersionedObjects(infos, clientConfig.GroupVersion.String(), kapi.Codecs.LegacyCodec(*clientConfig.GroupVersion))
+	var gv unversioned.GroupVersion
+	if clientConfig.GroupVersion != nil {
```

Comment:
This should never be `nil`.  We key the map based on `config.GrouopVersion.String()` in `factory.go`, so by the time this is returned to you it should always have a value.